dtyper.py

```diff
@@ -1,23 +1,39 @@
-from dataclasses import make_dataclass, field
-from functools import wraps
+from __future__ import annotations
+
 import inspect
+from dataclasses import field, make_dataclass
+from functools import wraps
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Callable, Optional, Type, TypeVar, Union
+
+    from typing_extensions import ParamSpec
+
+    P = ParamSpec("P")
+    R = TypeVar("R")
 
 __version__ = '0.9.0'
 
 
 @wraps(make_dataclass)
-def dataclass(typer_command, base=None, **kwargs):
-    def to_field(k, v, default):
-        return (k, v) if default is ... else (k, v, field(default=default))
+def dataclass(
+    typer_command: Callable[P, R], base: Optional[Type] = None, **kwargs
+) -> Callable[[Union[Type, Callable]], Type]:
 
     if base is not None:
         kwargs['bases'] = *kwargs.get('bases', ()), base
 
     kwargs.setdefault('cls_name', typer_command.__name__)
-    kwargs['fields'] = [to_field(*i) for i in _params(typer_command)]
+    kwargs['fields'] = [
+        (p.name, p.annotation)
+        if p.default is inspect.Parameter.empty
+        else (p.name, p.annotation, field(default=p.default))
+        for p in _fixed_signature(typer_command).parameters.values()
+    ]
 
     @wraps(typer_command)
-    def dataclass_maker(function_or_class):
+    def dataclass_maker(function_or_class: Union[Type, Callable]) -> Type:
```

Comment:
Thanks for adding all these types!

As you can see, I believe in types but as 75% of my organization's programming, I have eschewed formal type checking as "too much work for me".

Is there some CI I could add to test these to this little rep?